packages/libs/react-ui/src/components/Modal/Modal.stories.tsx

```diff
@@ -1,44 +1,63 @@
-import type { IModalProps } from '@components/Modal';
-import { ModalProvider } from '@components/Modal';
 import type { Meta, StoryObj } from '@storybook/react';
 import React from 'react';
-import { Content } from './StoryComponents';
+import { Button } from '../Button';
+import type { IModalProps } from '../Modal';
+import { Modal } from './Modal';
+import { ModalContent } from './StoryComponents';
+
+import { useOverlayTrigger } from 'react-aria';
+import { useOverlayTriggerState } from 'react-stately';
 
 const meta: Meta<{ title?: string } & IModalProps> = {
-  title: 'Layout/Modal',
+  title: 'Overlays/Modal',
   parameters: {
     docs: {
       description: {
         component:
-          'The component library exposes a `ModalProvider` and `useModal` hook that can be used with an element with id "modalportal" to display content in a modal.<br><br>To render a modal you need to add `<div id="modalportal" />` as the last child of the document body and wrap your content in the `ModalProvider` component. Then you can pass jsx and a title to the `renderModal` function from the `useModal` hook to render content in the modal.<br><br>See the code for this story for an example.',
-      },
-    },
-  },
-  argTypes: {
-    title: {
-      control: {
-        type: 'text',
+          'This is a generic modal component that can be used to render any content inside a modal., if you want to render a modal with a specific layout, you can use the `Dialog` component.',
       },
-      description: 'Title of the modal.',
     },
   },
 };
 
 export default meta;
-type Story = StoryObj<{ title?: string } & IModalProps>;
+type Story = StoryObj<IModalProps>;
 
 export const Primary: Story = {
   name: 'Modal',
-  args: {
-    title: 'Default Title',
-  },
-  render: ({ title }) => {
+  render: () => {
+    const state = useOverlayTriggerState({});
+    const { triggerProps, overlayProps } = useOverlayTrigger(
+      { type: 'dialog' },
+      state,
+    );
```

Comment:
For now, I think it's best to not expose the Modal component. Right now it requires some knowledge of react-aria so I think it would be best to expose a hook to provide wrap react-aria, but seems like over engineering for now

Reply:
yeah, this is a storybook feel free to remove it and also remove the export from the index file